### ISSUE TITLE
Incorrect predicted counts for models with negative amplitude.

### ISSUE THREAD

**Issue by LauraOlivera** [gammapy/gammapy#3700]
**Gammapy version**
`0.20.dev18+g735b25cd5.d20220112`

**Bug description**
When computing the predicted counts of a model that has negative amplitude, the "signal" is masked by this line

https://github.com/gammapy/gammapy/blob/3cd64fdb2c53c8e5c697a9b85ef8d0486bff0b76/gammapy/datasets/evaluator.py#L333-L337

The following snippet, with a negative source

```python3
from gammapy.datasets import MapDataset
from gammapy.modeling.models import SkyModel, PointSpatialModel, PowerLawSpectralModel
import matplotlib.pyplot as plt

dataset = MapDataset.read('$GAMMAPY_DATA/cta-1dc-gc/cta-1dc-gc.fits.gz')
 
spatial_model = PointSpatialModel(lon_0 = "0 deg", lat_0 = "0 deg", frame='galactic')
spectral_model = PowerLawSpectralModel(amplitude = "-1e-12 TeV-1 cm-2 s-1", index = 2)

model = SkyModel(spatial_model = spatial_model, spectral_model = spectral_model, name = 'gc negative')
model.apply_irf["psf_after_edisp"] = True

dataset = dataset.to_image()
dataset.models = model

dataset.npred().plot(add_cbar=True)
plt.show()
```
results in 
![image](https://user-images.githubusercontent.com/22521727/149830664-5c67169b-3a8c-4032-b26e-efecbff322c6.png)

**Expected behavior**
The right npred should have a negative excess in the middle, like
![image](https://user-images.githubusercontent.com/22521727/149830767-1db92d4f-cfa3-46f8-8f82-9f6f0f64b7fc.png)

This is an issue, for example when fitting a model where there is no source. The amplitude should be consistent with background fluctuations, which will some times be negative.

**Comment by adonath** [comment]
Thanks a lot @LauraOlivera, indeed there is an obvious issue! Especially I remember there we convergence issues with upper limits at high energies. The clipping was introduced here https://github.com/gammapy/gammapy/pull/2342, because we saw negative predicted counts, which should not happened either. I think the proper fix here might be to move the zero clipping to `dataset.npred()` instead.

**Comment by registerrier** [comment]
This is solved by #3704. Closing the issue.
